### PR TITLE
Bug fix/bts 169

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Fix BTS-169: Cost estimation for LIMIT nodes was wrong
https://arangodb.atlassian.net/browse/BTS-169

This is a cosmetic change, but should not have any effect on execution plans other than fixing the number of estimated items.

- [x] :hankey: Bugfix 
- [x] :book: CHANGELOG entry made
- [x] :muscle: The behavior in this PR was *manually tested*
- [x] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [ ] No backports required
- [x] Backports required for: 3.7

#### Related Information

- [x] GitHub issue / Jira ticket number: https://arangodb.atlassian.net/browse/BTS-169

### Testing & Verification

- [x] This PR adds tests that were used to verify all changes:
  - [ ] Added **Regression Tests**
  - [ ] Added new C++ **Unit Tests**
  - [x] Added new **integration tests** 
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11581/

### CODE REVIEW

File: arangod/Aql/ExecutionNode.cpp [hunk 1]
```diff
@@ -1761,9 +1761,22 @@ void LimitNode::toVelocyPackHelper(VPackBuilder& nodes, unsigned flags,
...
+  double const skipCost = 0.000001;
+
+  size_t estimatedNrItems = estimate.estimatedNrItems;
+  if (estimatedNrItems >= _offset) {
+    estimatedNrItems -= _offset;
+    estimate.estimatedCost += _offset * skipCost;
+  } else {
+    estimatedNrItems = 0;
+    estimate.estimatedCost += (_offset - estimatedNrItems) * skipCost;
+  }
```
[the remark this answers is not in the document]
Comment (jsteemann) [reply arangod/Aql/ExecutionNode.cpp:1777]: That is true... Thanks!

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11628/

File: tests/js/server/aql/aql-optimizer-costs.js [hunk 2]
```diff
@@ -53,6 +53,100 @@ function optimizerCostsTestSuite () {
...
+        [ 100000000000, 10000000000000, 0 ],
+      ];
+
+      tests.forEach(function(test) {
+        let plan = AQL_EXPLAIN(query, { offset: test[0], limit: test[1] });
+        let node = helper.findExecutionNodes(plan, "LimitNode")[0];
+
+        // number of nodes cannot be estimated properly due to function call
+        assertEqual("LimitNode", node.type);
+        assertEqual(test[2], node.estimatedNrItems);
```
[the remark this answers is not in the document]
Comment (jsteemann) [reply tests/js/server/aql/aql-optimizer-costs.js]: added tests for costs

**Comment by jsteemann** [comment]
Tests blue

**Review by gnusi — APPROVED** [review]
LGTM

**Pull request merged**